src/main/java/de/tum/in/www1/artemis/service/TextExerciseFeedbackService.java

```diff
@@ -0,0 +1,171 @@
+        log.debug("Using athena to generate (text exercise) feedback request: {}", textExercise.getId());
+
+        // athena takes over the control here
+        var submissionOptional = participationService.findTextExerciseParticipationWithLatestSubmissionAndResult(participation.getId()).findLatestSubmission();
+
+        if (submissionOptional.isEmpty()) {
+            throw new BadRequestAlertException("No legal submissions found", "submission", "noSubmission");
+        }
+        var submission = submissionOptional.get();
+
+        Result automaticResult = new Result();
+        automaticResult.setAssessmentType(AssessmentType.AUTOMATIC_ATHENA);
+        automaticResult.setRated(true);
+        automaticResult.setScore(0.0);
+        automaticResult.setSuccessful(null);
+        automaticResult.setSubmission(submission);
+        automaticResult.setParticipation(participation);
+        try {
+            this.resultWebsocketService.broadcastNewResult((Participation) participation, automaticResult);
+
+            log.debug("Submission id: {}", submission.getId());
+
+            var athenaResponse = this.athenaFeedbackSuggestionsService.orElseThrow().getTextFeedbackSuggestions(textExercise, (TextSubmission) submission, false);
+
+            List<Feedback> feedbacks = athenaResponse.stream().filter(individualFeedbackItem -> individualFeedbackItem.description() != null).map(individualFeedbackItem -> {
+                var feedback = new Feedback();
+                feedback.setText(individualFeedbackItem.title());
+                feedback.setDetailText(individualFeedbackItem.description());
+                feedback.setHasLongFeedbackText(false);
+                feedback.setType(FeedbackType.AUTOMATIC);
+                feedback.setCredits(individualFeedbackItem.credits());
+                return feedback;
+            }).toList();
+
+            double totalFeedbacksScore = 0.0;
+            for (Feedback feedback : feedbacks) {
+                totalFeedbacksScore += feedback.getCredits();
+            }
+            totalFeedbacksScore = totalFeedbacksScore / textExercise.getMaxPoints() * 100;
+            automaticResult.setSuccessful(true);
+            automaticResult.setCompletionDate(ZonedDateTime.now());
+
+            // Limit between 0 and 100%
+            if (totalFeedbacksScore < 0) {
+                automaticResult.setScore(0.0);
+            }
+            else if (totalFeedbacksScore > 100) {
+                totalFeedbacksScore = 100.0;
+            }
+            automaticResult.setScore(totalFeedbacksScore);
+
+            automaticResult = this.resultRepository.save(automaticResult);
+            this.resultService.storeFeedbackInResult(automaticResult, feedbacks, true);
+            submissionService.saveNewResult(submission, automaticResult);
+
+            // This will create a new submission without results, this is important so that tutor assessment works as it used to.
```

Comment:
This seams like a hacky solution. Submissions can have multiple results, e.g. programming exercises can have an automatic result as well as a manual assessment

Reply:
Well we wanted to have a new submission when AI Feedback is generated anyway, this doubles as an easy solution to not rewrite the TextAssessment.

Reply:
I reverted it to having a new submission be created only in the case that a student submit himself and the existing submission has a result. This means that submissions still work as they used to without athena feedback, that is the latest (only 1 in case of 0 results) submission gets edited.